### PR TITLE
Improve modflow

### PR DESCRIPTION
**Pull request by dbrakenhoff**

### CODE REVIEW

File: pastas_plugins/modflow/modflow.py [hunk 1]
```diff
@@ -83,12 +78,18 @@ def create_model(self) -> None:
             ncol=1,
             delr=1,
             delc=1,
-            top=1000,
-            botm=-1000,
+            top=1.0,
+            botm=0.0,
```
Comment (martinvonk) [inline pastas_plugins/modflow/modflow.py:82]: i don't know if this works with constant_d..

File: pastas_plugins/modflow/modflow.py [hunk 2]
```diff
@@ -162,16 +154,18 @@ def update_model(self, p: ArrayLike):
...
         self._gwf.name_file.write()
 
+    @functools.lru_cache(maxsize=5)
+    def _get_head(self, p):
+        self.update_model(p=p)
+        success, _ = self._simulation.run_simulation(silent=True)
+        if success:
+            return self._gwf.output.head().get_ts((0, 0, 0))[:, 1]
+        else:
+            raise Exception("Model run failed")
```
Comment (martinvonk) [inline pastas_plugins/modflow/modflow.py:164]: If the model does not converge during optimization we now have an issue / exception. That's why the zeroes were returned.

File: pastas_plugins/modflow/modflow.py (hunk 1 above)
Comment (dbrakenhoff) [reply pastas_plugins/modflow/modflow.py:82]: The way it's set up now modflow doesn't do anything with the constant. It just computes stuff relative to 0.0.

Comment (dbrakenhoff) [reply pastas_plugins/modflow/modflow.py:82]: And the model is confined now.

File: pastas_plugins/modflow/modflow.py (hunk 2 above)
Comment (dbrakenhoff) [reply pastas_plugins/modflow/modflow.py:164]: Fair point, I'll make it an option. Perhaps something we also want to report :).

**Pull request merged**